bigframes/core/nodes.py

```diff
@@ -65,6 +65,10 @@ def session(self):
             return sessions[0]
         return None
 
+    @functools.cached_property
+    def _node_hash(self):
```

Comment:
Let's document this (needed for cached_property and to avoid infinite loop)

Reply:
added comment explaining need for this hash impl